android/app/src/main/java/com/androidstartupperfapp/MainActivity.java

```diff
@@ -64,4 +72,24 @@ public boolean onSupportNavigateUp() {
     return NavigationUI.navigateUp(navController, appBarConfiguration)
       || super.onSupportNavigateUp();
   }
+
+  @Override
+  public void invokeDefaultOnBackPressed() {
+    super.onBackPressed();
+  }
+
+  @Override
+  public void onBackPressed() {
+    ReactFragment reactFragment = ReactFragmentInstanceManager.getReactFragment();
+
+    if (reactFragment.isVisible()) {
+      if (NavigationStateStore.getIndex() > 0) {
+        reactFragment.onBackPressed();
+      } else {
+        ReactFragmentInstanceManager.hideReactFragment();
+      }
+    } else {
+      super.onBackPressed();
+    }
+  }
```

Comment:
[R] Find some way to fix the [`ReactChoreographer.removeFromCallback`](https://github.com/facebook/react-native/blob/v0.66.4/ReactAndroid/src/main/java/com/facebook/react/modules/core/ReactChoreographer.java#L137-L147) problem. This part always emits the following error log in the following scenario:

```
Tried to remove non-existent frame callback
```

1. Go back all the way to first native Android screen from React Native screen.
2. Press Android hardware back button (no visual response, expected app to be put to background).
3. Press Android hardware back button (finally, a visual response, app put to background as expected).

The log should clue us into what's going on at the fragment / activity / application lifecycle whether it be React / Android.

Reply:
Debug stack trace:

```java
removeFrameCallback:144, ReactChoreographer (com.facebook.react.modules.core)
clearFrameCallback:334, NativeAnimatedModule (com.facebook.react.animated)
onHostDestroy:305, NativeAnimatedModule (com.facebook.react.animated)
onHostDestroy:305, ReactContext (com.facebook.react.bridge)
moveToBeforeCreateLifecycleState:768, ReactInstanceManager (com.facebook.react)
onHostDestroy:657, ReactInstanceManager (com.facebook.react)
onHostDestroy:671, ReactInstanceManager (com.facebook.react)
onHostDestroy:73, ReactDelegate (com.facebook.react)
onDestroy:108, ReactFragment (com.facebook.react)
performDestroy:3219, Fragment (androidx.fragment.app)
destroy:774, FragmentStateManager (androidx.fragment.app)
moveToExpectedState:350, FragmentStateManager (androidx.fragment.app)
executeOpsTogether:2180, FragmentManager (androidx.fragment.app)
removeRedundantOperationsAndExecute:2100, FragmentManager (androidx.fragment.app)
popBackStackImmediate:781, FragmentManager (androidx.fragment.app)
popBackStackImmediate:692, FragmentManager (androidx.fragment.app)
handleOnBackPressed:664, FragmentManager (androidx.fragment.app)
handleOnBackPressed:436, FragmentManager$1 (androidx.fragment.app)
onBackPressed:192, OnBackPressedDispatcher (androidx.activity)
onBackPressed:552, ComponentActivity (androidx.activity)
onBackPressed:92, MainActivity (com.androidstartupperfapp)
onKeyUp:3784, Activity (android.app)
dispatch:2866, KeyEvent (android.view)
dispatchKeyEvent:4090, Activity (android.app)
superDispatchKeyEvent:122, ComponentActivity (androidx.core.app)
dispatchKeyEvent:84, KeyEventDispatcher (androidx.core.view)
dispatchKeyEvent:140, ComponentActivity (androidx.core.app)
dispatchKeyEvent:599, AppCompatActivity (androidx.appcompat.app)
dispatchKeyEvent:59, WindowCallbackWrapper (androidx.appcompat.view)
dispatchKeyEvent:3068, AppCompatDelegateImpl$AppCompatWindowCallback (androidx.appcompat.app)
dispatchKeyEvent:59, WindowCallbackWrapper (androidx.appcompat.view)
dispatchKeyEvent:390, DecorView (com.android.internal.policy)
processKeyEvent:5947, ViewRootImpl$ViewPostImeInputStage (android.view)
onProcess:5815, ViewRootImpl$ViewPostImeInputStage (android.view)
deliver:5310, ViewRootImpl$InputStage (android.view)
onDeliverToNext:5367, ViewRootImpl$InputStage (android.view)
forward:5333, ViewRootImpl$InputStage (android.view)
forward:5485, ViewRootImpl$AsyncInputStage (android.view)
apply:5341, ViewRootImpl$InputStage (android.view)
apply:5542, ViewRootImpl$AsyncInputStage (android.view)
deliver:5314, ViewRootImpl$InputStage (android.view)
onDeliverToNext:5367, ViewRootImpl$InputStage (android.view)
forward:5333, ViewRootImpl$InputStage (android.view)
apply:5341, ViewRootImpl$InputStage (android.view)
deliver:5314, ViewRootImpl$InputStage (android.view)
onDeliverToNext:5367, ViewRootImpl$InputStage (android.view)
forward:5333, ViewRootImpl$InputStage (android.view)
forward:5518, ViewRootImpl$AsyncInputStage (android.view)
onFinishedInputEvent:5676, ViewRootImpl$ImeInputStage (android.view)
run:3179, InputMethodManager$PendingEvent (android.view.inputmethod)
invokeFinishedInputEventCallback:2721, InputMethodManager (android.view.inputmethod)
finishedInputEvent:2712, InputMethodManager (android.view.inputmethod)
onInputEventFinished:3156, InputMethodManager$ImeInputEventSender (android.view.inputmethod)
dispatchInputEventFinished:143, InputEventSender (android.view)
nativePollOnce:-1, MessageQueue (android.os)
next:335, MessageQueue (android.os)
loop:183, Looper (android.os)
main:7656, ActivityThread (android.app)
invoke:-1, Method (java.lang.reflect)
run:592, RuntimeInit$MethodAndArgsCaller (com.android.internal.os)
main:947, ZygoteInit (com.android.internal.os)
```